src/libos/src/interrupt/mod.rs

```diff
@@ -38,11 +38,15 @@ pub fn do_handle_interrupt(
 /// Broadcast interrupts to threads by sending POSIX signals.
 pub fn broadcast_interrupts() -> Result<usize> {
     let should_interrupt_thread = |thread: &&ThreadRef| -> bool {
-        // TODO: check Thread::sig_mask to reduce false positives
-        thread.process().is_forced_to_exit()
+        let should_interrupt = thread.process().is_forced_to_exit()
             || thread.is_forced_to_stop()
-            || !thread.sig_queues().read().unwrap().empty()
-            || !thread.process().sig_queues().read().unwrap().empty()
+            || !thread.process().sig_queues().read().unwrap().empty();
```

Comment:
I think sigmask should be checked for process signal queue.

Reply:
Done